conjure-core/src/main/java/com/palantir/conjure/parser/services/PathString.java

```diff
@@ -39,7 +39,7 @@
     /** Returns the well-formed path associated with this path definition. */
     public abstract Path path();
 
-    private static final Pattern SEGMENT_PATTERN = Pattern.compile("^[a-zA-Z][a-zA-Z0-9_-]*$");
+    private static final Pattern SEGMENT_PATTERN = Pattern.compile("^[a-zA-Z][a-zA-Z0-9._-]*$");
```

Comment:
after some digging, i presume this when actual parsing happens in preparation for IR compilation

assuming this was copy pasta, i can also refactor these into a single place. let me know if my assumption is wrong before i proceed

Reply:
It seems like we're doing identical validation. I think for consistency we should remove the parse time check and do it as part of the httpPathValidator